docs/whats-new/release-candidate-1.md

```diff
@@ -0,0 +1,786 @@
+
+<!-- markdownlint-disable MD033 -->
+<video controls src="media/waitfor.mp4" title="Title"></video>
+<!-- markdownlint-enable MD033 -->
+
+There are two methods exposed to wait for a resource:
+
+1. `WaitFor`: Wait for a resource to be ready before starting another resource.
+1. `WaitForCompletion`: Wait for a resource to complete before starting another resource.
+
+#### Resource health checks
+
+`WaitFor` uses health checks to determine if a resource is ready. If a resource does not have any health checks, the app host will wait for the resource to be in the "Running" state before starting the dependent resource.
+
+For resources that expose HTTP endpoints we have added a streamlined way to define a health check which polls a specific path for a HTTP 200 response.
+
+```C#
+var builder = DistributedApplication.CreateBuilder(args);
+
+var catalogApi = builder.AddContainer("catalog-api", "catalog-api")
+                 .WithHttpEndpoint(targetPort: 8080)
+                 .WithHttpHealthCheck("/healthz");
+
+builder.AddProject<Projects.WebApplication1>("store")
+    .WithReference(catalogApi.GetEndpoint("http"))
+    .WaitFor(catalogApi);
+
+builder.Build().Run();
+```
+
+The above example adds a health check to the `catalog-api` resource. The app host will wait for the health check to return a healthy status before starting the `store` resource. It'll determine that the resource is ready when the the /healthz endpoint returns a 200 status code.
+
+While `store` is waiting for `catalog-api` to become healthy, the resources in the dashboard will appear as:
+
+![Waiting for an unhealthy resource before starting](images/waiting-for-unhealthy-resource.png)
+
+The app host's health check mechanism builds upon the `IHealthChecksBuilder` implementation from the `Microsoft.Extensions.Diagnostics.HealthChecks` namespace.
+
+Health checks report data, which is displayed in the dashboard:
+
+![Health check details in the dashboard's resource details view](images/health-check-details.png)
+
+Creating a custom health check is simple, first define a custom health check, and then associate the health check name with the resource.
+
+```C#
+var builder = DistributedApplication.CreateBuilder(args);
+
+var healthyAfter = DateTime.Now.AddSeconds(20);
+builder.Services.AddHealthChecks().AddCheck(
+    "delay20secs",
+    () => DateTime.Now > healthyAfter ? HealthCheckResult.Healthy() : HealthCheckResult.Unhealthy()
+    );
+
+var cache = builder.AddRedis("cache")
+                   .WithHealthCheck("delay20secs");
+
+builder.AddProject<Projects.MyApp>("myapp")
+       .WithReference(cache).WaitFor(cache);
+```
+
+The above example shows using a custom health check to blow the start up of the dependent project resource until 20 seconds
+after the apphost starts running. The `WithHealthCheck(...)` mechanism provides a simple mechanism to assocate a resource
+with health checks that are already registered in the health checks system by name (`delay20secs`).
```

Comment:
I found "blow" a bit confusing, and started editing. Seems clearer to me, but I'm not sure.

```suggestion
The above example adds a health check that considers resources unhealthy until 20 seconds after the app host starts, after which it reports them as healthy. This health check is added to the `cache` resource. As `myapp` waits for `cache`, `myapp` won't start until those 20 seconds have elapsed.

The `AddCheck(...)` and `WithHealthCheck(...)` methods provide a simple mechanism to create health checks and associate them with specific resources.
```